### PR TITLE
Update crd versions

### PR DESCRIPTION
**Pull request by zewolfe**
This PR updates crds from upstream. It also changes the process with which this is done. It now uses `bash` and `kustomize` which more closely reflects the standard across other repos.

Issue - https://github.com/giantswarm/giantswarm/issues/22388

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,12 @@ and this project's packages adheres to [Semantic Versioning](http://semver.org/s
 
 ## [Unreleased]
 
+## [0.10.3] - 2022-07-06
```
[the remark this answers is not in the document]
Comment (whites11) [reply CHANGELOG.md:10]: In other words, you should only add entries in the `Unreleased` section

File: CHANGELOG.md [hunk 2]
```diff
@@ -100,7 +106,8 @@ and this project's packages adheres to [Semantic Versioning](http://semver.org/s
 
 - Add first version of the Azure AD Pod Identity App.
 
-[Unreleased]: https://github.com/giantswarm/azure-ad-pod-identity-app/compare/v0.10.2...HEAD
```
Comment (whites11) [inline CHANGELOG.md]: Same as above, this should happen only automatically and never done by hand

File: config/helm/input/mic-exception.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    component: mic
+---
+apiVersion: "aadpodidentity.k8s.io/v1"
+kind: AzurePodIdentityException
+metadata:
+  name: aks-addon-exception
+  namespace: kube-system
+spec:
+  podLabels:
+    kubernetes.azure.com/managedby: aks
```
Comment (whites11) [inline config/helm/input/mic-exception.yaml:18]: Do we need this? We will never have pods with such label

File: config/helm/transformers/common-labels.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+apiVersion: builtin
+kind: LabelTransformer
+metadata:
+  name: common-labels
+labels:
+  app.kubernetes.io/managed-by: '{{ .Release.Service }}'
+  app.kubernetes.io/version: '{{ .Chart.AppVersion }}'
+  helm.sh/chart: '{{ .Chart.Name }}'
+  app.kubernetes.io/name: '{{ template "aad-pod-identity.name" . }}'
+  app.kubernetes.io/instance: '{{ .Release.Name }}'
```
Comment (whites11) [inline config/helm/transformers/common-labels.yaml:11]: we might want to have the team label here `application.giantswarm.io/team`

File: config/helm/transformers/deployment-labels.yaml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+---
+apiVersion: builtin
+kind: LabelTransformer
+metadata:
+  name: deployment-labels
+labels:
+  app.kubernetes.io/name: '{{ template "aad-pod-identity.name" . }}'
+  app.kubernetes.io/instance: '{{ .Release.Name }}'
```
Comment (whites11) [inline config/helm/transformers/deployment-labels.yaml:8]: here missing `application.giantswarm.io/team` label as well

File: hacks/move-generated-crds.sh [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+        new_crd_file="$(echo "$crd_file" | cut -c50-)" # remove first 50 chars
+        echo "$new_crd_file"
+        mv "$crd_file" "$new_crd_file"
+    done
+    for crd_file in aadpodidentity.k8s.io_v1_azurepodidentityexception_*.yaml; do
+        new_crd_file="$(echo "$crd_file" | cut -c52-)" # remove first 52 chars
+        echo "$new_crd_file"
+        mv "$crd_file" "$new_crd_file"
+    done
+    cd ../../../../..
```
Comment (whites11) [inline hacks/move-generated-crds.sh:30]: I don't like this. If you need to go back to the original directory I suggest using pushd/popd https://opensource.com/article/19/8/navigating-bash-shell-pushd-popd

File: CHANGELOG.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (zewolfe) [reply CHANGELOG.md:10]: I had conflicts after a PR was merged so I had to pull from master. I believe there was a release that made these changes. Not sure why its highlighted though.

[the remark this answers is not in the document]
Comment (whites11) [reply CHANGELOG.md:10]: you can try to squash your commits.
```
git reset $(git merge-base master $(git rev-parse --abbrev-ref HEAD))
```

than git add, git commit and git push --force

File: config/helm/kustomization.yaml [hunk 7]
```diff
@@ -0,0 +1,37 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+namespace: '{{ .Release.Namespace }}'
+
+resources:
+  - input/deployment-rbac.yaml
+  - input/mic-exception.yaml
```
Comment (fiunchinho) [inline config/helm/kustomization.yaml:8]: Did we apply any modification to the files on the `input` folder? Could we use a link for those files in a release upstream?

File: helm/azure-ad-pod-identity-app/Chart.yaml [hunk 8]
```diff
@@ -1,8 +1,9 @@
 apiVersion: v1
-appVersion: 1.7.4
+appVersion: 1.8.9
 description: Deploy components for the azure-ad-pod-identity-app
 name: azure-ad-pod-identity-app
-version: [[ .Version ]]
+version: 0.0.1
+# version: [[ .Version ]]
```
Comment (fiunchinho) [inline helm/azure-ad-pod-identity-app/Chart.yaml:6]: I guess we still need `version: [[ .Version ]]` ?

**Review by whites11 — APPROVED** [review]

File: config/helm/kustomization.yaml (hunk 7 above)
Comment (zewolfe) [reply config/helm/kustomization.yaml:8]: Good shout. No mods were made. All things considered, i'll incur the technical debt and sort this out in a smaller PR. Sound fair?

Comment (fiunchinho) [reply config/helm/kustomization.yaml:8]: Perfect!

**Pull request merged**